cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/Executor.java

```diff
@@ -180,58 +182,94 @@ public void run() {
     }
 
     /**
-     * Start the actual execution of the proposals.
+     *
+     *
+     * @param expectedState Current expected state.
+     * @param nextState Next state to be in if the current state is the given expected state.
+     * @param moveRunnable If the current state is the given expected state, runnable to execute after state transition.
+     */
+    private void maybeMove(ExecutorState.State expectedState,
```

Comment:
I am wondering if this separate method is necessary. The previous logic seem simpler and easier to understand. Is there reason of having this separate method?